### PR TITLE
feat: refreshLayout

### PR DESCRIPTION
**Pull request by dongwoo-kim**
### Please check if the PR fulfills these requirements
- [x] It's submitted to right branch according to our branching model
- [x] It's right issue type on title
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] It does not introduce a breaking change or has description for the breaking change

### Description
- Added `refreshLayout`
- Added `heightResizable` options

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: src/dispatch/dimension.ts [hunk 1]
```diff
@@ -23,3 +23,20 @@ export function setOffsetTop({ dimension }: Store, offsetTop: number) {
...
+  const { autoWidth, fitToParentHeight } = dimension;
+  const { clientHeight, clientWidth, scrollTop, scrollLeft } = containerEl;
+  const { top, left } = containerEl.getBoundingClientRect();
+
+  setOffsetTop(store, top + scrollTop);
+  setOffsetLeft(store, left + scrollLeft);
+  setWidth(store, clientWidth, autoWidth);
+
+  if (fitToParentHeight) {
+    if (parentEl && parentEl.clientHeight !== clientHeight) {
```
Comment (seonim-ryu) [inline src/dispatch/dimension.ts]: if문 2개 합치면 안되나요??

File: src/view/container.tsx [hunk 2]
```diff
@@ -181,19 +183,9 @@ export class ContainerComp extends Component<Props> {
...
-      this.props.dispatch('setWidth', clientWidth, true);
-    }
-
-    if (fitToParentHeight) {
-      const { parentElement } = rootElement;
-      if (parentElement && parentElement.clientHeight !== clientHeight) {
-        this.props.dispatch('setHeight', parentElement.clientHeight);
-      }
-    }
+    this.props.dispatch('refreshLayout', this.el!, rootElement.parentElement!);
```
Comment (seonim-ryu) [inline src/view/container.tsx]: `rootElement.parentElement!` -> `this.props.rootElement.parentElement!` 이렇게 쓸 수 있겠네요

File: src/view/container.tsx [hunk 3]
```diff
@@ -218,6 +211,7 @@ export class ContainerComp extends Component<Props> {
     } = this.props;
     const style = { width: autoWidth ? '100%' : width };
     const attrs = { [dataAttr.GRID_ID]: gridId };
+    console.log('heightResizable', heightResizable);
```
Comment (seonim-ryu) [inline src/view/container.tsx]: 콘솔 제거요ㅎ

Comment (jung-han) [inline src/view/container.tsx]: console.log가 남아있습니다 ㅎㅎㅎ

File: src/store/dimension.ts [hunk 4]
```diff
@@ -9,12 +9,17 @@ type OptDimension = {
   summaryHeight?: number;
   domWidth: number;
   summaryPosition?: SummaryPosition;
-  scrollX: boolean;
-  scrollY: boolean;
   headerHeight: number;
 } & Pick<
   OptGrid,
-  'width' | 'rowHeight' | 'minRowHeight' | 'bodyHeight' | 'minBodyHeight' | 'scrollX' | 'scrollY'
+  | 'width'
```
Comment (seonim-ryu) [inline src/store/dimension.ts]: 여기에 `|` 들어가도 되는건가요??

**Comment by js87zz** [comment]
이견 없습니다. 고생하셨습니다!

File: src/store/dimension.ts (hunk 4 above)
Comment (dongwoo-kim) [reply src/store/dimension.ts]: Prettier가 일부러 바꿔주는 것 같네요. 보기 좋으라고 ㅎ

File: src/dispatch/dimension.ts (hunk 1 above)
Comment (dongwoo-kim) [reply src/dispatch/dimension.ts]: 아. 이거 합쳐야겠네요 ㅎ

**Comment by seonim-ryu** [comment]
[6/11] 리뷰 완료하였습니다ㅎ

File: src/view/container.tsx (hunk 2 above)
Comment (dongwoo-kim) [reply src/view/container.tsx]: 받아들이겠습니다 ㅎ

**Pull request merged**